internal/benchmarks/src/contract-interaction.bench.ts

```diff
@@ -16,55 +23,82 @@ describe('Contract Interaction Benchmarks', () => {
+
+      const { waitForResult } = await CounterContractFactory.deploy(wallet);
+      contract = (await waitForResult()).contract;
 
-  afterEach(() => {
-    cleanup();
+      const { waitForResult: waitForResultCallTestContract } =
+        await CallTestContractFactory.deploy(wallet);
+      callTestContract = (await waitForResultCallTestContract()).contract;
+    } else {
+      const launched = await launchTestNode({
+        contractsConfigs: [
+          { factory: CounterContractFactory },
+          { factory: CallTestContractFactory },
+        ],
+      });
+
+      cleanup = launched.cleanup;
+      contract = launched.contracts[0];
+      callTestContract = launched.contracts[1];
+      wallet = launched.wallets[0];
+    }
+  };
+
+  beforeAll(setupTestEnvironment);
+
+  afterAll(() => {
+    if (!isDevnet && cleanup) {
+      cleanup();
+    }
   });
 
-  bench('should successfully execute a contract read function', async () => {
+  runBenchmark('should successfully execute a contract read function', async () => {
     const tx = await contract.functions.get_count().call();
-
     const { value } = await tx.waitForResult();
-
-    expect(JSON.stringify(value)).toEqual(JSON.stringify(bn(0)));
+    expect(value).toBeDefined();
   });
 
-  bench('should successfully execute a contract multi call', async () => {
+  runBenchmark('should successfully execute a contract multi call', async () => {
+    const initialValue = 100;
     const tx = await contract
-      .multiCall([contract.functions.increment_counter(100), contract.functions.get_count()])
+      .multiCall([
+        contract.functions.increment_counter(initialValue),
+        contract.functions.get_count(),
+      ])
       .call();
-
     const { value } = await tx.waitForResult();
-
-    expect(JSON.stringify(value)).toEqual(JSON.stringify([bn(100), bn(100)]));
+    expect(value).toBeDefined();
   });
 
-  bench('should successfully write to a contract', async () => {
+  runBenchmark('should successfully write to a contract', async () => {
     const tx = await contract.functions.increment_counter(100).call();
     await tx.waitForResult();
   });
 
-  bench('should successfully execute a contract mint', async () => {
+  runBenchmark('should successfully execute a contract mint', async () => {
     const tx = await callTestContract.functions.mint_coins(TestAssetId.A.value, bn(100)).call();
-
     await tx.waitForResult();
   });
 
-  bench('should successfully execute a contract deploy', async () => {
+  runBenchmark('should successfully execute a contract deploy', async () => {
     const factory = new CounterContractFactory(wallet);
     const { waitForResult } = await factory.deploy();
     const { contract: deployedContract } = await waitForResult();
+    expect(deployedContract).toBeDefined();
+  });
 
+  bench('should successfully execute a contract deploy as blobs', async () => {
```

Comment:
How come this isn't a `runBenchmark`?

Reply:
Given that this is a longer operation, there is no need to do 10 iterations of it, that would not only be a lengthy benchmark but an expensive one on `devent`